### ISSUE TITLE
[BUG] Crashing when downloading

### ISSUE THREAD

**Issue by RSAStudioGames** [aliparlakci/bulk-downloader-for-reddit#499]
- [x] I am reporting a bug.
- [x] I am running the latest version of BDfR
- [x] I have read the [Opening an issue](https://github.com/aliparlakci/bulk-downloader-for-reddit/blob/master/docs/CONTRIBUTING.md#opening-an-issue)
  
## Description
The program will download for a little while and then crash with the error message I pasted below. Ive tried it on different accounts and machines and it still has the same issue. Usually happens after about 50 - 60 downloads. However, I can easily recreate it on the same machine/acc just by running the command again and it will stop at the exact same post. I have also tried unsaving the last post before the crash and it just ends on the one before it.

## Command
```
python -m bdfr download PATH --user me --saved --authenticate --search-existing
```

## Environment (please complete the following information):
 - OS: [e.g. Windows 10]
 - Python version: [e.g. 3.9.4]

## Logs
```
Traceback (most recent call last):
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\bdfr\__main__.py", line 152, in <module>
    cli()
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\click\decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\bdfr\__main__.py", line 81, in cli_download
    reddit_downloader.download()
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\bdfr\downloader.py", line 45, in download
    self._download_submission(submission)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\bdfr\downloader.py", line 73, in _download_submission
    content = downloader.find_resources(self.authenticator)
  File "C:\Users\RSAStudioGames\AppData\Local\Programs\Python\Python39\lib\site-packages\bdfr\site_downloaders\gallery.py", line 23, in find_resources
    image_urls = self._get_links(self.post.gallery_data['items'])
TypeError: 'NoneType' object is not subscriptable
```

**Comment by aliparlakci** [comment]
Please provide full log output as it was mentioned in the contribution guidelines. Tracebacks themselves don't provide enough information to debug.

**Comment by RSAStudioGames** [comment]
I figured it out I think. There was a delete post that contained a gallery and that seemed to have caused an error.

**Comment by Serene-Arc** [comment]
@aliparlakci @RSAStudioGames I think that this bug was fixed in 77aaee9 which is live in the development branch. If you checkout that, this bug will be fixed.
